lib/Authentication/Core/MerchantConfiguration.php

```diff
@@ -1033,31 +1102,117 @@ public function setDisableRequestMLEForMandatoryApisGlobally($disableRequestMLEF
     }
 
     /**
-     * Get the value of mapToControlMLEonAPI
+     * Get the value of mapToControlMLEonAPI (raw user-supplied)
      *
-     * @return array<string, bool>
+     * @return array<string,string|bool>
      */
     public function getMapToControlMLEonAPI()
     {
         return $this->mapToControlMLEonAPI;
     }
 
     /**
-     * Set the value of mapToControlMLEonAPI
+     * Set the value of mapToControlMLEonAPI.
+     * Accepts associative array or stdClass whose values are string|bool.
      *
-     * @param array<string, bool> $mapToControlMLEonAPI
+     * Parsing to internal request/response maps happens immediately.
+     *
+     * @param array<string,string|bool>|object $mapToControlMLEonAPI
+     * @return void
+     * @throws AuthException if validation fails
      */
     public function setMapToControlMLEonAPI($mapToControlMLEonAPI)
     {
+        if ($mapToControlMLEonAPI === null) {
+            return;
+        }
+
+        if (is_object($mapToControlMLEonAPI)) {
+            $mapToControlMLEonAPI = get_object_vars($mapToControlMLEonAPI);
+        }
+
+        if (!is_array($mapToControlMLEonAPI)) {
+            $error_message = "mapToControlMLEonAPI must be an associative array or object.";
+            $exception = new AuthException($error_message, 0);
+            if (self::$logger) { self::$logger->error($error_message); }
+            throw $exception;
+        }
+
+        foreach ($mapToControlMLEonAPI as $k => $v) {
+            if (!is_string($k)) {
+                $error_message = "mapToControlMLEonAPI keys must be strings.";
+                $exception = new AuthException($error_message, 0);
+                if (self::$logger) { self::$logger->error($error_message); }
+                throw $exception;
+            }
+            if (!is_string($v) && !is_bool($v)) {
+                $error_message = "mapToControlMLEonAPI values must be string or bool.";
+                $exception = new AuthException($error_message, 0);
+                if (self::$logger) { self::$logger->error($error_message); }
+                throw $exception;
+            }
+        }
+
+        // Validate the map value format - allowed values are true::true, false::false, ::true, true::, ::false, false::, true, false
         if ($mapToControlMLEonAPI !== null) {
-            if (is_array($mapToControlMLEonAPI) && $this->isAssocArrayOfStringBool($mapToControlMLEonAPI)) {
-                $this->mapToControlMLEonAPI = $mapToControlMLEonAPI;
-            } else {
-                throw new InvalidArgumentException("mapToControlMLEonAPI in merchantConfig must be an array<string, bool> type.");
+            $this->validateMapToControlMLEonAPIValues($mapToControlMLEonAPI);
+        }
+        
+        $this->mapToControlMLEonAPI = $mapToControlMLEonAPI;
+        
+        // Populate internal maps from the main map
+        if ($mapToControlMLEonAPI !== null) {
```

Comment:
If you have already set `$this->mapToControlMLEonAPI`, why are you using the parameter `$mapToControlMLEonAPI` here?

Review this and see which one is the source of truth.

Either use the parameter and set `$this->mapToControlMLEonAPI` later, OR use `$this->mapToControlMLEonAPI` only.